### PR TITLE
Use argon2 as default over bcrypt.

### PR DESCRIPTION
**Pull request by 7hoenix**
Why is this change necessary?
------------------------------

When this generator was first created Bcrypt was still the OWASP recommended default choice https://github.com/dashbitco/mix_phx_gen_auth_demo/pull/1#discussion_r403110955

That has [since changed](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html) and argon2 is now the officially recommended choice, with bcrypt only being recommended if argon2 is not available.

This PR swaps the default from bcrypt to argon2.

Should we also default Windows to argon2? <- I don't know and so I left it.

<Details>

I generated a sample phx project and configured it to look at this local copy of phx_gen_auth.

Screen Shot 2021-04-22 at 7 08 44 AM

Ran the command with no `hashing-lib` specified.

Screen Shot 2021-04-22 at 7 09 02 AM

It uses argon2 lib.

Screen Shot 2021-04-22 at 7 10 24 AM

The tests are green.

</Details>

### CODE REVIEW

File: lib/mix/phx_gen_auth/hashing_library.ex [hunk 1]
```diff
@@ -3,38 +3,38 @@ defmodule Mix.Phx.Gen.Auth.HashingLibrary do
 
   defstruct [:module, :mix_dependency, :test_config]
 
-  def build("bcrypt") do
```
Comment (josevalim) [inline lib/mix/phx_gen_auth/hashing_library.ex]: Can you please revert the changes to this file? We don't need to reorder these clauses, do we? :)

**Comment by josevalim** [comment]
Thank you! We should go ahead with this change indeed. I have added a comment. Also, the build is failing. Can you plaese take a look at the tests?

File: test/mix/tasks/phx.gen.auth_test.exs [hunk 2]
```diff
@@ -84,7 +84,7 @@ defmodule Mix.Tasks.Phx.Gen.AuthTest do
       )
 
       assert_file("config/test.exs", fn file ->
-        assert file =~ "config :bcrypt_elixir, :log_rounds, 1"
+        assert file =~ "config :argon2_elixir, :log_rounds, 1"
```
Comment (teamon) [inline test/mix/tasks/phx.gen.auth_test.exs:87]: Based on my experience and [`argon2_elixir` docs](https://github.com/riverrun/argon2_elixir/blob/master/lib/argon2/stats.ex#L29), this should be:

```elixir
config :argon2_elixir, t_cost: 1, m_cost: 8
```

**Comment by MaSven** [comment]
As for the windows question. Citate from comeonin

> Argon2 is currently considered to be the strongest password hashing function, and it is the one we recommend.
> Bcrypt and Pbkdf2 are viable alternatives, but they are less resistant than Argon2, to attacks using GPUs or dedicated hardware.

So this superseedes now pbkdf2. Maybe we should also change the default on windows.

File: test/mix/tasks/phx.gen.auth_test.exs (hunk 2 above)
Comment (goncalotomas) [reply test/mix/tasks/phx.gen.auth_test.exs:87]: I'm guessing this would also need to be changed in `test/mix/tasks/phx_gen_auth/integration_tests/default_app_test.exs`

**Comment by aaronrenner** [comment]
Thanks for this PR @7hoenix!. Now that phx_gen_auth has been merged into phoenix 1.6, I'm archiving this project and all phx.gen.auth issues will need to be handled in the phoenix repo.
